android/app/src/main/java/com/tgmu/tgmu/ui/fragment/ExperiencesFragment.kt

```diff
@@ -1,13 +1,162 @@
 package com.tgmu.tgmu.ui.fragment
 
 import android.os.Bundle
+import android.util.Log
 import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.core.widget.addTextChangedListener
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.google.android.material.search.SearchView
+import com.google.android.material.snackbar.Snackbar
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
 import com.tgmu.tgmu.R
+import com.tgmu.tgmu.databinding.FragmentDiscoverBinding
+import com.tgmu.tgmu.databinding.FragmentExperiencesBinding
+import com.tgmu.tgmu.ui.adapters.CompactExperienceAdapter
+import com.tgmu.tgmu.ui.adapters.MovieSearchSuggestionsAdapter
+import com.tgmu.tgmu.ui.viewmodel.ExperienceViewModel
+import com.tgmu.tgmu.ui.viewmodel.MoviesViewModel
+import com.tgmu.tgmu.utils.Constants
+import com.tgmu.tgmu.utils.Resource
 import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 
 @AndroidEntryPoint
 class ExperiencesFragment : Fragment(R.layout.fragment_experiences) {
+    private var _binding: FragmentExperiencesBinding? = null
+    private val binding get() = _binding!!
+    private val experienceViewModel: ExperienceViewModel by viewModels()
+    private val moviesViewModel: MoviesViewModel by viewModels()
```

Comment:
Change to by activityViewModel since the view model is shared inside the activity between multiple fragments